src/Engine/ProtoCore/ClassTable.cs

```diff
@@ -386,20 +388,50 @@ public ProcedureNode GetDisposeMethod()
 
     public class ClassTable
     {
+        
+        private List<ClassNode> classNodes = new List<ClassNode>();
+
+        //Symbol table to manage symbols with namespace
+        private Namespace.SymbolTable symbolTable = new Namespace.SymbolTable();
+        
+        private List<ClassNode> GetClassHierarchy(ClassNode node)
+        {
+            var cNodes = new List<ClassNode> {node};
+            
+            while (node.Base != Constants.kInvalidIndex)
+            {
+                node = ClassNodes[node.Base];
+                cNodes.Add(node);
+            }
+            return cNodes;
+        }
+
+        private ClassNode GetCommonBaseClass(IEnumerable<Symbol> symbols)
+        {
+            var cNodes = ClassNodes.Where(node => symbols.Any(s => s.Id == node.ID));
+            var baseLists = new List<List<ClassNode>>();
+            foreach (var classNode in cNodes)
+            {
+                var baseNodes = GetClassHierarchy(classNode);
+                baseLists.Add(baseNodes);
+            }
+            var arr = baseLists.ToArray();
+
+            if (!arr.Any()) return null;
+
+            var baseClass = ArrayUtils.GetCommonItems(arr);
```

Comment:
is there a case where this could be a list which actually contains 2 items?

Reply:
I don't think there will be a case where this will be a list of more than one item but I'm just returning the first item just in case.